### PR TITLE
refactoring in preparation for unauthenticated access and website

### PR DESCRIPTION
**Pull request by saanobhaai**

### CODE REVIEW

**Comment by saanobhaai** [comment]
[Add KBA analysis](https://trello.com/c/2ieBp33y/168-add-kba-analysis)

File: src/api/migrations/0010_auto_20220329_1441.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    RestorationStats,
+    SCL,
+    SCLStats,
+    SurveyLandscape,
+    SurveyStats
+)
+
+
+def truncate(*args, **kwargs):
+    RestorationLandscape.objects.all().delete()
```
Comment (gridcell) [inline src/api/migrations/0010_auto_20220329_1441.py]: Less likely for future migration problems if you use 
```
from django.apps import apps
RestorationLandscape = apps.get_model("api.RestorationLandscape")
...

RestorationLandscape.objects.all().delete()
```

File: src/api/models/scl.py [hunk 2]
```diff
@@ -78,31 +81,19 @@ def __str__(self):
         name = ""
         if hasattr(self, "name"):
             name = " {}".format(self.name)
-        return _("%s%s %s [%s]") % (self.ee_name, name, self.species, self.date)
+        return _("%s %s%s %s [%s]") % (self.ee_name, self.lsid, name, self.species, self.date)
```
Comment (gridcell) [inline src/api/models/scl.py:84]: Mix of string formatting (f-string ?)

File: src/api/reports/stats.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def calc_landscape_stats(country, date, species):
         conservation_landscape_stats=calc_conservation_landscape_stats(
             country, date, species
         ),
-        fragment_landscape_stats=calc_fragment_landscape_stats(country, date, species),
+        # fragment_landscape_stats=calc_fragment_landscape_stats(country, date, species),
```
Comment (gridcell) [inline src/api/reports/stats.py:48]: Is this commented out field still needed?

File: src/api/reports/stats.py [hunk 4]
```diff
@@ -60,13 +60,13 @@ def calc_conservation_landscape_stats(country, date, species):
...
 
 
-def calc_fragment_landscape_stats(country, date, species):
-    filter_args = dict(
-        fragment__date__lte=date, fragment__species_id=species, country=country
-    )
-
-    qs = FragmentStats.objects.filter(**filter_args)
-    return _calc_summary(qs, date_field="fragment__date")
+# def calc_fragment_landscape_stats(country, date, species):
```
Comment (gridcell) [inline src/api/reports/stats.py:63]: Still needed?

File: src/api/reports/views.py [hunk 5]
```diff
@@ -9,11 +9,11 @@
 from . import species_report
 from . import stats
 from ..models import Species
-from ..resources.base import UserCountryPermission
+# from ..resources.base import UserCountryPermission
```
Comment (gridcell) [inline src/api/reports/views.py]: Still needed?

File: src/api/resources/fragment_stats.py [hunk 6]
```diff
@@ -1,46 +1,89 @@
...
-                return FragmentStats.objects.none()
-
-        filters = {
-            "country": self.request.query_params["country"],
-            "fragment__species": self.request.query_params["fragment__species"],
-            "fragment__date": self.request.query_params["fragment__date"],
-        }
-
-        return FragmentStats.objects.filter(**filters).select_related()
+        return SurveyFragmentStats.objects.select_related()
```
Comment (gridcell) [inline src/api/resources/fragment_stats.py]: can move this to a property

`queryset = SurveyFragmentStats.objects.select_related()`

File: src/api/resources/base.py [hunk 7]
```diff
@@ -96,4 +96,4 @@ def destroy(self, request, *args, **kwargs):
 
 class StatsViewSet(BaseAPIViewSet):
     pagination_class = GeoJsonPagination
-    permission_classes = [UserCountryPermission]
+    # permission_classes = [UserCountryPermission]
```
Comment (gridcell) [inline src/api/resources/base.py]: needed?

File: src/api/resources/indigenous_stats.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        fields = ["country", "species"]
+
+
+class IndigenousRangeStatsViewSet(StatsViewSet):
+    serializer_class = IndigenousRangeStatsSerializer
+    filter_class = IndigenousRangeStatsFilterSet
+    ordering_fields = ["species", "country"]
+
+    def get_queryset(self):
+        return IndigenousRangeCountryStats.objects.select_related()
```
Comment (gridcell) [inline src/api/resources/indigenous_stats.py]: can move to queryset property

File: src/api/resources/restorationls_stats.py [hunk 9]
```diff
@@ -40,30 +41,8 @@ class RestorationStatsViewSet(StatsViewSet):
...
-            "restoration_landscape__species": self.request.query_params[
-                "restoration_landscape__species"
-            ],
-            "restoration_landscape__date": self.request.query_params[
-                "restoration_landscape__date"
-            ],
-        }
-
-        return RestorationStats.objects.filter(**filters).select_related()
+        return RestorationStats.objects.select_related()
```
Comment (gridcell) [inline src/api/resources/restorationls_stats.py]: property

File: src/api/resources/scl_stats.py [hunk 10]
```diff
@@ -31,44 +24,24 @@ class Meta(StatsSerializer.Meta):
...
 
     @action(detail=False, methods=["get"])
     def available_dates(self, request):
-        if (
-            "country" not in self.request.query_params
-            or "scl__species" not in self.request.query_params
-        ):
-            return SCLStats.objects.none()
+        filters = {}
+        if request.query_params.get("country"):
```
Comment (gridcell) [inline src/api/resources/scl_stats.py]: can set these `get` calls to a variable before if statement

File: src/api/reports/stats.py (hunk 3 above)
Comment (saanobhaai) [reply src/api/reports/stats.py:48]: All the reports stuff is going to get redone anyway; I'm leaving in for Alex to see how it used to work

**Pull request merged**